File: python/src/unittest/python/WordDocumentTest.py

from WordDocument import *
from Format import *
from BackImage import *
from Image import *
from License import *
import unittest


class WordDocumentTest(unittest.TestCase):

    def setUp(self):
        self.word_doc = WordDocument(1)

    def testBackground(self):
        self.assertEquals(BackgroundImage().getImage().get_url(),
                          self.word_doc.getBackground().get_url())

    def testOfficeVersion(self):
        self.word_doc.setMsOffice(2.0)
        self.assertEquals(2.0, self.word_doc.getOfficeVersion())

    def testLicense(self):
        self.assertEquals(1, self.word_doc.getOfficeLicense().getLicSerialNum())

    def testInvalidLicense(self):
        self.assertFalse(self.word_doc.
                         licenceInvalidCannotEdit())


    def testFormat(self):
        self.assertEquals(Font('Arial', 11).getFontText(),
                          self.word_doc.get_format()
                          .getFont().getFontText())

File: java/src/test/java/mx/iteso/WordDocument_Test.java
package mx.iteso;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;

import org.junit.Test;

/**
 * Word Document Test Class.
 */
public class WordDocument_Test {

    /**
     * Word document.
     */
    private WordDocument wordDocument = new WordDocument(2000);

    /**
     * Set sharing permission test.
     */
    @Test
    public void testMSOfficeVersion() {
        wordDocument.setMSOffice(2.1f);
        assertEquals("2.1", Float.toString(wordDocument.getMSOffice()));
    }

    /**
     * License test.
     */
    @Test
    public void testLicense() {
        wordDocument.setMSLicense(new MSLicense(2000));
        assertEquals(2000, wordDocument.getMSLicense().getLicenseNumber());
    }

    /**
     * License test.
     */
    @Test
    public void testRestrictedEdit() {
        assertFalse(
                wordDocument.restricInvalidLicense(new MSLicense(2000)));
    }

    /**
     * Font evaluator.
     */
    @Test
    public void testFormat() {
        wordDocument.setFormat(new Format());
        assertEquals("Roboto", wordDocument.getFormat().getFont().getFontName());
    }

    /**
     * Background test.
     */
    @Test
    public void testgetBackground() {
        wordDocument.setBackground(new Image("image.png"));
        assertEquals("image.png", wordDocument.getBackground().getUrl());
    }



}
